doc/getting_started/build_sw.md

```diff
@@ -1,217 +1,335 @@
+To simplify the installation of Bazel, and provide a means to seamlessly update the Bazel version we use in the future, we provide a shell script that acts as a wrapper for invocations of "`bazel ...`".
+To use it, you two options:
+1. use "`./bazelisk.sh ...`" instead of "`bazel ...`" to invoke of Bazel subcommands, or
+1. set the following alias (e.g., in your `.bashrc` file) to accomplish the same:
+```console
+alias bazel="$REPO_TOP/bazelisk.sh"
+```
 
-`./meson_init.sh` itself is idempotent, but this behavior can be changed with additional flags; see `./meson_init.sh` for more information.
-For this reason, most examples involving Meson will include a call to `./meson_init.sh`, but you will rarely need to run it more than once per checkout.
+### Manual Installation
 
-Building an executable `foo` destined to run on the OpenTitan device `$DEVICE` will output the following files under `build-bin/sw/device`:
-* `foo_$DEVICE.elf`: the linked program, in ELF format.
-* `foo_$DEVICE.bin`: the linked program, as a plain binary with ELF debug information removed.
-* `foo_$DEVICE.dis`: the disassembled program with inline source code.
-* `foo_$DEVICE.vmem`: a Verilog memory file which can be read by `$readmemh()` in Verilog code.
+This section is optional and can be skipped if you completed the instructions above in _Automatic Installation_.
 
-In general, this executable is built by the `foo_export_$DEVICE` target.
-For example, this builds the `pwrmgr_smoketest` test binary for DEVICE `sim_dv`:
+While the automatic installation is convenient, by installing Bazel directly, you can get some quality of life features like tab completion.
+If you haven't yet installed Bazel, and would like to, you can add it to apt and install it on Ubuntu systems with the following commands [as described in the Bazel documentation](https://bazel.build/install/ubuntu):
 
 ```console
-ninja -C build-out sw/device/tests/pwrmgr_smoketest_export_sim_dv
+sudo apt install apt-transport-https curl gnupg
+curl -fsSL https://bazel.build/bazel-release.pub.gpg | gpg --dearmor > bazel.gpg
+sudo mv bazel.gpg /etc/apt/trusted.gpg.d/
+echo "deb [arch=amd64] https://storage.googleapis.com/bazel-apt stable jdk1.8" |
+sudo tee /etc/apt/sources.list.d/bazel.list
+sudo apt update && sudo apt install bazel-4.2.0
 ```
 
-Building an executable destined to run on a host machine (i.e., under `sw/host`) will output a host excecutable under `build-bin/sw/host`, which can be run directly.
-
-### Troubleshooting the build system
+or by following [instructions for your system](https://bazel.build/install).
 
-If you encounter an error running `./meson_init.sh` you could re-run using the `-f` flag which will erase any existing building directories to yield a clean build.
-This sledgehammer is only intended to be used as a last resort when the existing configuration is seriously broken:
+## Building Software with Bazel
 
+Running
```

Comment:
where might be the right place to mention the `DISABLE_VERILATOR_BUILD` flag?

Reply:
ah i see it down below.

Reply:
i think it's fine as is. There's probably no "good" place to put it, so it's own separate section is good.